src/main/scala/is/hail/methods/MendelErrors.scala

```diff
@@ -74,47 +70,36 @@ object MendelErrors {
     if (nSamplesDiscarded > 0)
       warn(s"$nSamplesDiscarded ${ plural(nSamplesDiscarded, "sample") } discarded from .fam: sex of child is missing.")
 
-    val sampleTrioRoles = mutable.Map.empty[String, List[(Int, Int)]]
-    trios.zipWithIndex.foreach { case (t, ti) =>
-      sampleTrioRoles += (t.kid -> sampleTrioRoles.getOrElse(t.kid, List.empty[(Int, Int)]).::(ti, 0))
-      sampleTrioRoles += (t.knownDad -> sampleTrioRoles.getOrElse(t.knownDad, List.empty[(Int, Int)]).::(ti, 1))
-      sampleTrioRoles += (t.knownMom -> sampleTrioRoles.getOrElse(t.knownMom, List.empty[(Int, Int)]).::(ti, 2))
-    }
+    val trioMatrix = vds.trioMatrix(Pedigree(trios), completeTrios = true)
+    val rowType = trioMatrix.rowType
+    val nTrios = trioMatrix.nSamples
 
     val sc = vds.sparkContext
-    val sampleTrioRolesBc = sc.broadcast(sampleTrioRoles)
     val triosBc = sc.broadcast(trios)
     // all trios have defined sex, see filter above
     val trioSexBc = sc.broadcast(trios.map(_.sex.get))
 
-    val zeroVal: MultiArray2[GenotypeType] = MultiArray2.fill(trios.length, 3)(NoCall)
-
-    def seqOp(a: MultiArray2[GenotypeType], s: Annotation, g: Genotype): MultiArray2[GenotypeType] = {
-      sampleTrioRolesBc.value.get(s.asInstanceOf[String]).foreach(l => l.foreach { case (ti, ri) => a.update(ti, ri, Genotype.gtType(g)) })
-      a
-    }
-
-    def mergeOp(a: MultiArray2[GenotypeType], b: MultiArray2[GenotypeType]): MultiArray2[GenotypeType] = {
-      for ((i, j) <- a.indices)
-        if (b(i, j) != NoCall)
-          a(i, j) = b(i, j)
-      a
-    }
-
     new MendelErrors(vds.hc, vds.vSignature, trios, vds.stringSampleIds,
-      vds
-        .aggregateByVariantWithKeys(zeroVal)(
```

Comment:
This should be the only use of `aggregateByVariantWithKeys`.  You should be able to delete all teh `aggregateByVariant*` methods.